From 6c60131fbcce3b9889d726a238726db9a09bda3a Mon Sep 17 00:00:00 2001
From: Kazunori Kobayashi <kkobayas@igel.co.jp>
Date: Mon, 16 Apr 2012 17:24:20 +0900
Subject: [PATCH 23/97] ext/dfbvideosink: change the order of color formats
 for the caps negotiation

This patch changes color format lists obtained by get caps function.
This change puts a higher priority on the following formats.
- RGB32
- UYVY
---
 ext/directfb/dfbvideosink.c |   26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/ext/directfb/dfbvideosink.c b/ext/directfb/dfbvideosink.c
index 2c1d808..ddd536b 100644
--- a/ext/directfb/dfbvideosink.c
+++ b/ext/directfb/dfbvideosink.c
@@ -1220,14 +1220,14 @@ gst_dfbvideosink_getcaps (GstBaseSink * bsink)
       if (dfbvideosink->vio) {
         caps = gst_caps_new_empty ();
         gst_caps_append (caps,
-            gst_dfbvideosink_get_caps_from_format (DSPF_RGB16));
-        gst_caps_append (caps,
-            gst_dfbvideosink_get_caps_from_format (DSPF_RGB24));
-        gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_RGB32));
         gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_UYVY));
         gst_caps_append (caps,
+            gst_dfbvideosink_get_caps_from_format (DSPF_RGB16));
+        gst_caps_append (caps,
+            gst_dfbvideosink_get_caps_from_format (DSPF_RGB24));
+        gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_YV12));
         gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_ARGB));
@@ -1251,6 +1251,15 @@ gst_dfbvideosink_getcaps (GstBaseSink * bsink)
        * DSPF_UYVY
        * DSPF_YV12
        */
+      if (dfbvideosink->vio) {
+        gst_caps_append (caps,
+            gst_dfbvideosink_get_caps_from_format (DSPF_RGB32));
+      }
+      if (gst_dfbvideosink_can_blit_from_format (dfbvideosink, DSPF_UYVY,
+              accelerated) || dfbvideosink->vio) {
+        gst_caps_append (caps,
+            gst_dfbvideosink_get_caps_from_format (DSPF_UYVY));
+      }
       if (gst_dfbvideosink_can_blit_from_format (dfbvideosink, DSPF_RGB16,
               accelerated) || dfbvideosink->vio) {
         gst_caps_append (caps,
@@ -1261,10 +1270,6 @@ gst_dfbvideosink_getcaps (GstBaseSink * bsink)
         gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_RGB24));
       }
-      if (dfbvideosink->vio) {
-        gst_caps_append (caps,
-            gst_dfbvideosink_get_caps_from_format (DSPF_RGB32));
-      }
       if (gst_dfbvideosink_can_blit_from_format (dfbvideosink, DSPF_ARGB,
               accelerated) || dfbvideosink->vio) {
         gst_caps_append (caps,
@@ -1275,11 +1280,6 @@ gst_dfbvideosink_getcaps (GstBaseSink * bsink)
         gst_caps_append (caps,
             gst_dfbvideosink_get_caps_from_format (DSPF_YUY2));
       }
-      if (gst_dfbvideosink_can_blit_from_format (dfbvideosink, DSPF_UYVY,
-              accelerated) || dfbvideosink->vio) {
-        gst_caps_append (caps,
-            gst_dfbvideosink_get_caps_from_format (DSPF_UYVY));
-      }
       if (gst_dfbvideosink_can_blit_from_format (dfbvideosink, DSPF_I420,
               accelerated)) {
         gst_caps_append (caps,
-- 
1.7.10.4

